### PR TITLE
lokiexporter: Remove dependency on https://github.com/grafana/loki, due to incompatible dep issues

### PR DESCRIPTION
**Pull request by gramidt**
**Description:**
Removed dependency on https://github.com/grafana/loki due to incompatible dependency issues. The Grafana/loki project has many dependencies that are not needed by this exporter, and a few of those dependencies have additional dependencies that are incompatible with other components within this project. Given that the "replace" directive is only honored within the root go.mod, these sub-dependencies break other components and prevent this exporter from successfully being added to the binary.

The [loki-client-go](https://github.com/grafana/loki-client-go) project was recently started to help address the dependency requirements and make Loki clients easier to implement, but is marked experimental and has not released a version yet. Once this project has released a supported version, we will evaluate switching to it (#2386).

To help progress and overcome these issues immediately, this PR has copied the Loki log protos from ( https://github.com/grafana/loki/tree/v2.1.0/pkg/logproto ), with this decision being documented within the committed [exporter/lokiexporter/logproto/README.md](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/38b239d15fdfb7a6b4d6e9417e4b0ffbe9b831cd/exporter/lokiexporter/logproto/README.md). 

**Link to tracking Issue:**
#1894 

**Testing:**
Unit Tests

**Documentation:**
READMEs

### CODE REVIEW

File: exporter/lokiexporter/logproto/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Notice
+
+This code was copied from https://github.com/grafana/loki/tree/master/pkg/logproto, due to incompatible dependency 
+issues when this exporter was added to the top level binary (top level go.mod replace, even with a specific version,
+is not a good options due to other components' dependency requirements).
+
```
Comment (bogdandrutu) [inline exporter/lokiexporter/logproto/README.md:6]: Would it be possible to extract this code in a independent module on the loki project in the future to remove this duplicated code and ensure nice dependency?

**Review by bogdandrutu — APPROVED** [review]

File: exporter/lokiexporter/logproto/README.md (hunk 1 above)
Comment (bogdandrutu) [reply exporter/lokiexporter/logproto/README.md:6]: Saw the comment, please file an issue and assign to you so we can track progress on the removing of this code?

Comment (gramidt) [reply exporter/lokiexporter/logproto/README.md:6]: Absolutely, @bogdandrutu! However, due to my permissions, I'll either need your or someone else to assign the issue to me. I also have a few other issues that need assigned and labelled for tracking.

Comment (gramidt) [reply exporter/lokiexporter/logproto/README.md:6]: @bogdandrutu - https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/2386

**Pull request merged**